### PR TITLE
Move user cluster resource handling from the cluster controller to the UCCM

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
This PR moves almost all user cluster resource handling from the cluster controller to the user cluster controller.

Things which are currently still kept inside the cluster controller:
- OpenVPN client cert
  Let the user cluster controller generate it? Or generate it in the cluster controller and just propagate it via the user cluster controller?

Regardless of the above mentioned open issue, this PR works(According to the tests) and can be merged.

**Release note**:
```release-note
NONE
```

### CODE REVIEW

**Comment by mrIncompetent** [comment]
Sidenote:
I'll fix the enqueue mechanism of the UCCM in a follow up PR

File: api/pkg/controller/usercluster/reconciler.go [hunk 1]
```diff
@@ -120,24 +161,125 @@ func (r *reconciler) reconcileRoles(ctx context.Context) error {
 	return nil
 }
 
-func (r *reconciler) reconcileServiceAcconts(ctx context.Context) error {
-	creators := []resources.NamedServiceAccountCreatorGetter{
-		userauth.ServiceAccountCreator(),
+func (r *reconciler) reconcileRoleBindings(ctx context.Context) error {
+	// kube-system
```
Comment (alvaroaleman) [inline api/pkg/controller/usercluster/reconciler.go:165]: leftover?

File: api/pkg/controller/usercluster/reconciler.go [hunk 2]
```diff
@@ -120,24 +161,125 @@ func (r *reconciler) reconcileRoles(ctx context.Context) error {
...
+		scheduler.RoleBindingAuthDelegator(),
+		controllermanager.RoleBindingAuthDelegator(),
+	}
+	if err := resources.ReconcileRoleBindings(creators, metav1.NamespaceSystem, r.Client, r.cache); err != nil {
+		return fmt.Errorf("failed to reconcile Roles in kube-system Namespace: %v", err)
 	}
 
-	if err := resources.ReconcileServiceAccounts(creators, metav1.NamespaceSystem, r.Client, r.cache); err != nil {
-		return fmt.Errorf("failed to reconcile ServiceAccounts in the namespace %s: %v", metav1.NamespaceSystem, err)
+	// kube-public
```
Comment (alvaroaleman) [inline api/pkg/controller/usercluster/reconciler.go:176]: Leftover?

File: api/pkg/controller/usercluster/reconciler.go [hunk 3]
```diff
@@ -120,24 +161,125 @@ func (r *reconciler) reconcileRoles(ctx context.Context) error {
...
+	creators = []resources.NamedRoleBindingCreatorGetter{
+		machinecontroller.KubePublicRoleBindingCreator(),
+		machinecontroller.ClusterInfoAnonymousRoleBindingCreator(),
+	}
+	if err := resources.ReconcileRoleBindings(creators, metav1.NamespacePublic, r.Client, r.cache); err != nil {
+		return fmt.Errorf("failed to reconcile Roles in kube-public Namespace: %v", err)
+	}
+
+	// Default
```
Comment (alvaroaleman) [inline api/pkg/controller/usercluster/reconciler.go:185]: Leftover?

File: api/pkg/controller/usercluster/reconciler.go (hunk 1 above)
Comment (mrIncompetent) [reply api/pkg/controller/usercluster/reconciler.go:165]: Nope, just a comment stating that the following RoleBindings will be created in the `kube-system` namespace.
Afaik all of them are still required.

File: api/pkg/controller/usercluster/reconciler.go (hunk 2 above)
Comment (mrIncompetent) [reply api/pkg/controller/usercluster/reconciler.go:176]: https://github.com/kubermatic/kubermatic/pull/3086/files#r270883358 should apply here as well

File: api/pkg/controller/usercluster/reconciler.go (hunk 3 above)
Comment (mrIncompetent) [reply api/pkg/controller/usercluster/reconciler.go:185]: https://github.com/kubermatic/kubermatic/pull/3086/files#r270883358 should apply here as well

**Review by kron4eg — APPROVED** [review]
/approve

**Pull request merged**